pkg/asset/manifests/network.go

```diff
@@ -145,40 +129,43 @@ func (no *Networking) Load(f asset.FileFetcher) (bool, error) {
 	return false, nil
 }
 
-// generateCustomCnoConfig generates the defaultNetwork for Cluster Network Operator
-// configuration, and returns the byte data with Cluster Network Operator manifest.
-func (no *Networking) generateCustomCnoConfig(defaultNetwork *operatorv1.DefaultNetworkDefinition) ([]byte, error) {
-	if defaultNetwork == nil {
-		return nil, errors.New("defaultNetwork must be specified")
-	}
-	dnConfig := operatorv1.Network{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: operatorv1.SchemeGroupVersion.String(),
-			Kind:       "Network",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "cluster",
-		},
-		Spec: operatorv1.NetworkSpec{
-			OperatorSpec:   operatorv1.OperatorSpec{ManagementState: operatorv1.Managed},
-			DefaultNetwork: *defaultNetwork,
-		},
+// clusterNetworkOperatorConfig conditionally generates the operatorv1.Networking config if customizations
+// are needed, otherwise it is omitted to fallback to default behavior.
+func clusterNetworkOperatorConfig(ic *installconfig.InstallConfig, cns []configv1.ClusterNetworkEntry, sn []string) (*operatorv1.Network, error) {
+	var cnoCfg *operatorv1.Network
+	var err error
+	switch ic.Config.Platform.Name() {
+	case aws.Name:
+		cnoCfg, err = generateCustomNetworkConfigMTU(ic, cns, sn)
```

Comment:
Maybe we can simplify all of this a bit here. It appears that ic.Config.NetworkType != "OVNKubernetes", cnoCfg can be nil. So, we can start off clusterNetworkOperatorConfig() with that check so that it does not need to be checked multiple times in the nested mentods.

For 
`case aws.Name:
        cnoCfg = ovnNetworkOperatorConfig(cns, sn)
        cnoCfg.Spec.DefaultNetwork.OVNKubernetesConfig.MTU = generateCustomNetworkConfigMTU()`
       
 Can we rework   generateCustomNetworkConfigMTU() to just return an `int` MTU?
 That way the AWS and vSphere cases would look consistent.